search/2016-09-01/swagger/searchindex.json

```diff
@@ -0,0 +1,289 @@
+        }
+      },
+      "description": "Status of an indexing operation for a single document.",
+      "x-ms-external": true
+    },
+    "DocumentIndexResult": {
+      "properties": {
+        "value": {
+          "x-ms-client-name": "Results",
+          "type": "array",
+          "readOnly":  true,
+          "items": {
+            "$ref": "#/definitions/IndexingResult"
+          },
+          "description": "The list of status information for each document in the indexing request."
+        }
+      },
+      "description": "Response containing the status of operations for all documents in the indexing request."
+    },
+    "IndexActionType": {
+      "type": "string",
+      "enum": [
+        "upload",
+        "merge",
+        "mergeOrUpload",
+        "delete"
+      ],
+      "x-ms-enum": { "name": "IndexActionType" },
+      "description": "Specifies the operation to perform on a document in an indexing batch."
+    },
+    "SearchMode": {
+      "type": "string",
+      "enum": [
+        "any",
+        "all"
+      ],
+      "x-ms-enum": { "name": "SearchMode" },
+      "description": "Specifies whether any or all of the search terms must be matched in order to count the document as a match."
+    },
+    "QueryType": {
+      "type": "string",
+      "enum": [
+        "simple",
+        "full"
+      ],
+      "x-ms-enum": { "name": "QueryType" },
+      "description": "Specifies the syntax of the search query. The default is 'simple'. Use 'full' if your query uses the Lucene query syntax."
+    },
+    "SearchParametersPayload": {
+      "properties": {
+        "count": {
+          "type": "boolean",
+          "description": "A value that specifies whether to fetch the total count of results. Default is false. Setting this value to true may have a performance impact. Note that the count returned is an approximation."
+        },
+        "facets": {
+          "externalDocs": {
+            "url": "https://docs.microsoft.com/rest/api/searchservice/Search-Documents"
+          },
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of facet expressions to apply to the search query. Each facet expression contains a field name, optionally followed by a comma-separated list of name:value pairs."
+        },
+        "filter": {
+          "externalDocs": {
+            "url": "https://docs.microsoft.com/rest/api/searchservice/OData-Expression-Syntax-for-Azure-Search"
+          },
+          "type": "string",
+          "description": "The OData $filter expression to apply to the search query."
+        },
+        "highlight": {
+          "type": "string",
```

Comment:
@dsgouda I have some (IMO) well-grounded objections to regex constraints. Rather than repeat them, I'll direct you to [this discussion](https://github.com/Azure/azure-rest-api-specs/pull/637) on an earlier PR.

Regarding min/max constraints, I believe they are valuable for documentation purposes where applicable. It may be the case that we have no limits on some of these properties right now, but if we do, they are documented in the (hand written) REST API reference. However, I would still refrain from adding them to the Swagger until AutoRest supports [disabling client-side validation](https://github.com/Azure/autorest/issues/1583).